awx/ui/client/src/instance-groups/instances/instances.controller.js

```diff
@@ -1,20 +1,79 @@
-export default ['$scope', 'InstanceList', 'GetBasePath', 'Rest', 'Dataset','Find', '$state', '$q',
-    function($scope, InstanceList, GetBasePath, Rest, Dataset, Find, $state, $q) {
-        let list = InstanceList;
+function InstancesController ($scope, $state, $http, models, Instance, strings, Dataset) {
+    const { instanceGroup } = models;
+    const vm = this || {};
+    vm.strings = strings;
+    vm.panelTitle = instanceGroup.get('name');
+    vm.instances = instanceGroup.get('related.instances.results');
+    vm.instance_group_id = instanceGroup.get('id');
 
-        init();
+    init();
 
-        function init(){
-            $scope.optionsDefer = $q.defer();
-            $scope.list = list;
-            $scope[`${list.iterator}_dataset`] = Dataset.data;
-            $scope[list.name] = $scope[`${list.iterator}_dataset`].results;
+    function init() {
+        $scope.list = {
+            iterator: 'instance',
+            name: 'instances'
+        };
+        $scope.collection = {
+            basepath: 'instances',
+            iterator: 'instance'
+        };
+        $scope[`${$scope.list.iterator}_dataset`] = Dataset.data;
+        $scope[$scope.list.name] = $scope[`${$scope.list.iterator}_dataset`].results;
+    }
+
+    vm.tab = {
+        details: {
+            _go: 'instanceGroups.edit',
+            _params: { instance_group_id: vm.instance_group_id }
+        },
+        instances: {
+            _active: true,
+            _go: 'instanceGroups.instances',
+            _params: { instance_group_id: vm.instance_group_id }
+        },
+        jobs: {
+            _go: 'instanceGroups.jobs',
+            _params: { instance_group_id: vm.instance_group_id }
         }
+    };
+
+    vm.toggle = (toggled) => {
+        let instance = _.find(vm.instances, 'id', toggled.id);
+        instance.enabled = !instance.enabled;
 
-        $scope.isActive = function(id) {
-            let selected = parseInt($state.params.instance_id);
-            return id === selected;
+        let data = {
+            "enabled": instance.enabled
         };
 
-    }
-];
\ No newline at end of file
+        let req = {
+            method: 'PUT',
+            url: instance.url,
+            data
+        };
+
+        $http(req).then(vm.onSaveSuccess);
```

Comment:
does this need error handling?  Also should this use some sort of model abstraction?